pkg/controller/expansion/stats_reporter.go

```diff
@@ -71,20 +67,28 @@ func (r *etRegistry) report(ctx context.Context) {
 		return
 	}
 
+	if r.statusReport == nil {
+		r.statusReport = make(map[metrics.Status]int64)
+	}
+
 	totals := make(map[metrics.Status]int64)
 	for _, status := range r.cache {
 		totals[status]++
 	}
 
 	for _, s := range metrics.AllStatuses {
-		ctx, err := tag.New(ctx, tag.Insert(statusKey, string(s)))
-		if err != nil {
-			log.Error(err, "failed to create status tag for expansion templates")
-			continue
-		}
-		err = metrics.Record(ctx, etM.M(totals[s]))
-		if err != nil {
-			log.Error(err, "failed to record total expansion templates")
-		}
+		r.statusReport[s] = totals[s]
+	}
+}
+
+func (r *etRegistry) registerCallback() error {
+	_, err := meter.RegisterCallback(r.observeETM, etM)
+	return err
+}
+
+func (r *etRegistry) observeETM(_ context.Context, o metric.Observer) error {
+	for s, v := range r.statusReport {
```

Comment:
I'm pretty sure every `observe` function will require a read mutex. LMK if you see an exception to this